impress-org/givewp
# Feature/paypal donations billing field support 5744

Resolves #5744 

## Description

The donor does not billing fields to add billing details when donating with PayPal donations. This pull request adds support for billing fields. You will get option to activate billing field in `Settings ->  Payment Gateways ->  PayPal -> PayPal Donations`.

## Affects

This pull request affects only PayPal donations donation processing logic.

Note: Donor address will add to Paypal order but not with a subscription. PayPal order asks for [payer](https://developer.paypal.com/docs/api/orders/v2/#definition-payer) address while subscription asks for the [subscribe](https://developer.paypal.com/docs/api/subscriptions/v1/#definition-subscriber)r shipping address.

## Visuals

![image](https://user-images.githubusercontent.com/1784821/116430374-6931cf00-a864-11eb-8273-fc8be3d8402f.png)
![image](https://user-images.githubusercontent.com/1784821/116430303-58815900-a864-11eb-8d7d-7c9df64f0d80.png)

## Testing Instructions

The visual change happens only on the frontend when you enable or disable the `Collect Billing Details` setting. This pull request considers merging if the billing address appears on the receipt, store into the GiveWP database and admin can view billing info on the donation details screen.
Test with pull request with PayPal `Express account` and `PPCP` account.

## Pre-review Checklist
